src/common/utils/iter.ts

```diff
@@ -23,3 +23,67 @@ export function* take<T>(src: Iterable<T>, n: number): Iterable<T> {
     break outer;
   }
 }
+
```

Comment:
Just a note: would make sense to add tests for these utils if they are used in many places.

Reply:
I can do that in a separate PR, but good idea.